### PR TITLE
Add recipe manual testing to DG

### LINKED ISSUE #206: Update DG
@Imranr2 
- [x] Add acknowledgements 
- [x] Remove setting up, getting started

### Design
- [x] Architecture
- [x] UI component
- [x] Logic component
- [x] Model component
- [x] Storage component

### Implementation
- [x] List feature sequence diagram - remove UI

### Documentation, Logging...
- [x] Make it relevant to our current app

### User Story
- [x] Update to match current app

### Use Cases
- [x] Add edit order, recipe and statistics

### NFR & Glossary
- [x] Add stuff

### Manual Testing (Numbers are based on UG)
- [x] 4.1 to 4.3 @Imranr2 
- [x] 4.4 and 4.8 @EuniceSim142 
- [x] 4.5.1 to 4.5.6 @sivayogasubramanian 
- [ ] 4.5.7 to 4.6.3 @Manoharan-Ajay-Anand 
- [x] 4.6.4 to 4.6.7, 4.9, 4.10 @zognin

### PR DESCRIPTION
**Pull request by Manoharan-Ajay-Anand**
Closes #206 . See DG [here](https://manoharan-ajay-anand.github.io/tp/DeveloperGuide.html).

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
        <br>Expected: No change in the order list display. Error details shown in the result display box.
 
+1. Listing all orders
+    1. Prerequisite: Order list has at least one order.
+    1. Test case: `list-o`
+        <br>Expected: Order list displays all orders.
+    1. Test case: `list-o abc`
+       <br>Expected: Order list displays all orders.
+
+1. Marking an order as done
```
Comment (sivayogasubramanian) [inline docs/DeveloperGuide.md:1608]: Why no negative/failing test cases for this?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+            * Order Deadline: 12 October 2021 at 12 pm
+            * Order Quantity: 1
+            * Completion status: Completed
+    1. Test case: `done-o 1`
+       <br>Expected: The first order is marked as completed. It's position in the Order list may change. Details of
+           marked order is shown in the result display.
+    1. Test case: `done-o 2`
+       <br>Expected: The second order is still marked as completed. Details of marked order is shown in the result display.
+
+1. Marking an order as undone
```
Comment (sivayogasubramanian) [inline docs/DeveloperGuide.md:1636]: Same as above

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case: `undone-o 1`
+       <br>Expected: The first order is still marked as completed. Details of marked order is shown in the result display.
+
 ### Recipe
 
+1. Adding a Recipe
+    1. Prerequisites (should be satisfied for each individual test case):
+        1. Recipe list does not contain a recipe with the following details:
+            * Recipe Name: Chicken Rice
+            * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
```
Comment (sivayogasubramanian) [inline docs/DeveloperGuide.md:1670]: is `whole` and `Whole` different? Your commands all use `Whole`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+            * Recipe Price: $4.00
+    1. Test case: `add-r rn/Chicken Rice ri/Rice-200-g, Chicken-1-Whole rp/4`
+        <br> Expected: A new recipe is added to the recipe list with the given details. Details of the added
+        recipe are shown in the result display box.
+    1. Test case: `add-r rn/Chicken Rice ri/Rice-200-g, Chicken-1-Whole`
+       <br> Expected: No recipe is added. Error details shown in the result display box.
+    1. Test case: `add-r rn/Chicken Rice rp/4`
```
Comment (sivayogasubramanian) [inline docs/DeveloperGuide.md:1677]: This is a valid command!!!!

Please check all your commands if it is valid. If this kind of bugs are reported in the PE, we are really shooting ourselves in the foot.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+            * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+            * Order Price: $4.00
+            * Order Deadline: 12 October 2021 at 12 pm
+            * Order Quantity: 1
+            * Completion status: Completed
+    1. Test case: `undone-o 2`
+       <br> The second order is marked as uncompleted. It's position in the Order list may change. Details of
+       marked order is shown in the result display.
+    1. Test case: `undone-o 1`
+       <br>Expected: The first order is still marked as completed. Details of marked order is shown in the result display.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1662]:
```suggestion
       <br>Expected: The first order is still marked as uncompleted. Details of marked order are shown in the result display.
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+            * Client Address: Blk 30 Geylang Street 29. #06-40
+            * Recipe Name: Chicken Rice
+            * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+            * Order Price: $4.00
+            * Order Deadline: 12 October 2021 at 12 pm
+            * Order Quantity: 1
+            * Completion status: Completed
+    1. Test case: `done-o 1`
+       <br>Expected: The first order is marked as completed. It's position in the Order list may change. Details of
+           marked order is shown in the result display.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1632]:
```suggestion
       <br>Expected: The first order is marked as completed. It's position in the Order list may change. Details of
           marked order are shown in the result display.
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case: `done-o 2`
+       <br>Expected: The second order is still marked as completed. Details of marked order is shown in the result display.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1634]:
```suggestion
       <br>Expected: The second order is still marked as completed. Details of marked order are shown in the result display.
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -1598,8 +1598,129 @@ testers are expected to do more *exploratory* testing.
...
+            * Client Address: Blk 30 Geylang Street 29. #06-40
+            * Recipe Name: Chicken Rice
+            * Recipe Ingredients: Rice x 200 g, Chicken x 1 whole
+            * Order Price: $4.00
+            * Order Deadline: 12 October 2021 at 12 pm
+            * Order Quantity: 1
+            * Completion status: Completed
+    1. Test case: `undone-o 2`
+       <br> The second order is marked as uncompleted. It's position in the Order list may change. Details of
+       marked order is shown in the result display.
```
Comment (Imranr2) [inline docs/DeveloperGuide.md:1660]:
```suggestion
       <br> The second order is marked as uncompleted. It's position in the Order list may change. Details of
       marked order are shown in the result display.
```

**Review by Imranr2 — COMMENTED** [review]
LGTM! I'll let @sivayogasubramanian review and merge

**Review by sivayogasubramanian — APPROVED** [review]

**Pull request merged**